### DISCUSSION TITLE
Sunrise/sunset times by conditions

### DISCUSSION THREAD

**Discussion by Macintosh-Fan** [jpb10/SolarCalculator#17] [Q&A] [answered]
I assume the current sunrise/sunset times calculated are when the observer is at sea level and the temperature is 20°C, along with other atmospheric things. What could be cool is provide parameters for calculations based on these conditions. It might be a bit too much, but just an idea if possible.

EDIT: it seems like this feature already exists, so I converted the category to Q&A.

**Comment by jpb10** [comment, marked as answer]
You can account for all these factors by providing a custom `altitude` argument. 

The default value of -0.8333 degrees corrects for the Sun's half-diameter and refraction at the horizon.

Example sketch [SunriseSunsetAltitude](https://github.com/jpb10/SolarCalculator/tree/main/examples/SunriseSunsetAltitude) shows how to correct for the dip in the horizon (at a height above sea level).

You are free to use any model applicable to your location and weather.
